src/components/CharacterAssignment.jsx

```diff
@@ -0,0 +1,229 @@
+          // Reassign characterIds to the unassigned column
+            const reassignedIds = newColumns[key].characterIds;
+            const reassignedCounts = newColumns[key].lineCount;
+            newColumns.unassignedCharacters.characterIds = [...newColumns.unassignedCharacters.characterIds, ...reassignedIds];
+            newColumns.unassignedCharacters.lineCount += reassignedCounts;
+            const index = newOrder.findIndex(id => id === key);
+            newOrder.splice(index, 1);
+          }
+        };
+      }
+      //Update the new column and order values after the actor changes
+      setColumns(newColumns)
+      setColumnOrder(newOrder)
+    }
+
+    // Initial setup of actor columns, no script is selected
+    else {
+    const actorColumns = {
+      'unassignedCharacters': {
+        id:'unassignedCharacters',
+        title: 'Unassigned Characters',
+        characterIds: [],
+        lineCount: 0
+      }
+    };
+    const actorIds = ['unassignedCharacters']
+    actors.forEach((actor) => {
+      actorColumns[actor.id] = {
+        id: actor.id.toString(),
+        title: actor.name,
+        characterIds: [],
+        lineCount: 0
+      }
+      actorIds.push(actor.id.toString())
+    })
+      setColumns(actorColumns)
+      setColumnOrder(actorIds)
+    }
+  }
+
+
+  // Gets called after a character is dropped on a new actor or unassigned. Updates the value in the database.
+  const addCharacter = async (characterId, actorId, currentScript) => {
+    // Removes actor assignment if the character is moved back to unassigned
+    if (actorId === 'unassignedCharacters'){
+      actorId = null;
+    }
+    try {
+      const response = await fetch(`/api/script/${currentScript}/characters`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({characterId, actorId})
+      })
+      if (!response.ok) {
+        console.error(`server error: ${response.body} when assigning the character to the actor`);
+      }
+    } catch (error) {
+      console.error(`network error: ${error} when assigning the character to the actor`);
+      }
+  }
+
+
+  // Update values after a character has been dropped in a column
+  const onDragEnd = (result) => {
+    const { destination, source, draggableId } = result
+    // No destination, change nothing
+    if (!destination) {
+      return;
+    }
+    // Dropped back in same location, change nothing
+    if (destination.droppableId === source.droppableId) {
+      return;
+    }
+    // Get the column ids of the source and destination columns
+    const start = columns[source.droppableId];
+    const finish = columns[destination.droppableId]
+
+    // Character is in a different list, move it and update both column values
+    const startCharacterIds = Array.from(start.characterIds)
+    const startLineCount = start.lineCount -= characters[draggableId].lineCount
```

Comment:
this is fine, but one way to be more defensive/robust with this coding is to calculate the total lines by adding up every actor in the list whenever this value is needed. That could avoid the line count getting out of sync permanently if there is some issue with updating the count correctly _once_ and then staying that way.